### PR TITLE
Use vite.js instead of webpack

### PR DESCRIPTION
**Pull request by will-moore**
Replace webpack with vite.js.

This also updates all dependencies to latest versions.

To test:
 - check that parade works as usual
 - Just try a basic filter -> table -> plot workflow

### CODE REVIEW

**Comment by will-moore** [comment]
Failing build with:

```
Error: Package: 2:nodejs-18.16.0-1nodesource.x86_64 (nodesource)
           Requires: libstdc++.so.6(GLIBCXX_3.4.20)(64bit)
Error: Package: 2:nodejs-18.16.0-1nodesource.x86_64 (nodesource)
           Requires: libstdc++.so.6(GLIBCXX_3.4.21)(64bit)
Error: Package: 2:nodejs-18.16.0-1nodesource.x86_64 (nodesource)
           Requires: libm.so.6(GLIBC_2.27)(64bit)
Error: Package: 2:nodejs-18.16.0-1nodesource.x86_64 (nodesource)
           Requires: libstdc++.so.6(CXXABI_1.3.9)(64bit)
Error: Package: 2:nodejs-18.16.0-1nodesource.x86_64 (nodesource)
           Requires: libc.so.6(GLIBC_2.28)(64bit)
```

**Comment by will-moore** [comment]
This is causing failure in merge-ci web deploy: see https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-web/1634/console
because it needs a different version of node.js (same as failures above).

--exclude

**Comment by will-moore** [comment]
Just for the record, this PR (migration from webpack to vite.js) was helped in a big way by chatGPT as follows:

Screenshot 2023-04-19 at 13 33 35

Screenshot 2023-04-19 at 13 33 47

Screenshot 2023-04-19 at 13 33 57

Screenshot 2023-04-19 at 13 34 07

Screenshot 2023-04-19 at 13 34 24

File: .omeroci/app-deps [hunk 1]
```diff
@@ -3,5 +3,5 @@
 set -e
 set -u
 yum -y install epel-release
-curl --silent --location https://rpm.nodesource.com/setup_12.x | sudo bash -
+curl --silent --location https://rpm.nodesource.com/setup_14.x | sudo bash -
```
Comment (jburel) [inline .omeroci/app-deps:6]: you tried with 18 and it fails
16 is the closest LTS so  I think we should try with 16

**Comment by will-moore** [comment]
The text says OMERO.web is required. I was going to put a version of omero-web, and then I realised that this is already specified in setup.py (as are all dependencies) and I didn't want to duplicate it.
I'm not sure saying "This requires omero-web" adds much and "This requires `omero-web>=5.6.1`" is duplication, so I think it's cleaner without?

**Comment by jburel** [comment]
I was thinking something like
```
OMERO.figure
============

An OMERO.web app for creating figures from images in OMERO.

For full details see `SUPPORT.md <https://github.com/ome/omero-figure/blob/master/SUPPORT.md>`_.

Requirements
------------

* OMERO.web 5.6.0 or newer.

```
I think it is easier to quickly identify what is what

**Comment by will-moore** [comment]
Updated README as suggested

**Review by jburel — APPROVED** [review]

**Comment by joshmoore** [comment]
Latest builds since June 21st are failing.

e.g. https://latest-ci.openmicroscopy.org/jenkins/job/OMERO-web/1365/console

```
00:15:06     > omero_parade@0.1.3 build /tmp/pip-req-build-_2trlcwo
00:15:06     > vite build
00:15:06     
00:15:06     (node:46956) ExperimentalWarning: The ESM module loader is experimental.
00:15:06     file:///tmp/pip-req-build-_2trlcwo/node_modules/vite/bin/vite.js:7
00:15:06         await import('source-map-support').then((r) => r.default.install())
00:15:06         ^^^^^
00:15:06     
00:15:06     SyntaxError: Unexpected reserved word
00:15:06         at Loader.moduleStrategy (internal/modules/esm/translators.js:81:18)
00:15:06         at async link (internal/modules/esm/module_job.js:37:21)
00:15:06     npm ERR! code ELIFECYCLE
00:15:06     npm ERR! errno 1
00:15:06     npm ERR! omero_parade@0.1.3 build: `vite build`
00:15:06     npm ERR! Exit status 1
00:15:06     npm ERR!
00:15:06     npm ERR! Failed at the omero_parade@0.1.3 build script.
00:15:06     npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Is this related? Does something need updating in the build?

**Comment by will-moore** [comment]
Ah, yes - I think latest-ci infrastructure hasn't been updated as merge-ci has.

**Pull request merged**